js/ui/viewSelector.js

```diff
@@ -284,7 +303,19 @@ const ViewsDisplay = new Lang.Class({
         Main.overview.addAction(clickAction, false);
         this._searchResults.actor.bind_property('mapped', clickAction, 'enabled', GObject.BindingFlags.SYNC_CREATE);
 
-        this.actor = new ViewsDisplayContainer(this.entry, this._allView, this._searchResults);
+        let icon = new St.Icon({ icon_name: 'edit-find-symbolic',
+                                style_class: 'discovery-feed-icon',
+                                icon_size: 16,
+                                track_hover: true });
+        this.gc = new St.Button({ style_class: 'discovery-feed',
+                                  x_align: Clutter.ActorAlign.CENTER,
+                                  y_align: Clutter.ActorAlign.CENTER });
+        this.gc.set_child(icon);
+        this.gc.connect('clicked', Lang.bind(this, function() {
+          Main.discoveryFeed.show(0);
```

Comment:
See comment above about encapsulating a lot of this logic into its own subclass of `St.Button`